protocol/contracts/governance/ElderElection.sol

```diff
@@ -0,0 +1,143 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./Templar.sol";
+
+/**
+ * @title A contract recording voting in the election of temple elders.
+ *
+ * @notice
+ * The election runs "continuously" - timing for when the results are actually
+ * actioned is handled at the social/policy layer.
+ * 
+ * Any address can vote by endorsing candidates. However only endorsements from voters
+ * who hold temple (aka "token stakeholders") or who are temple team members (aka "impact
+ * stakeholders") have vote value.
+ *
+ * EIP712 structured data signing supports 'gasless voting' via a relay
+ */
+contract ElderElection is EIP712, AccessControl {
+
+    bytes32 public constant CAN_NOMINATE = keccak256("CAN_NOMINATE");
+
+    using Counters for Counters.Counter;
+    using SafeERC20 for IERC20;
+
+    /// @notice The NFT ids (ie discord ids) of the Templars that are candidates for
+    /// the ElderElection
+    mapping(uint256 => bool) public candidates;
+
+    /// @notice The number of active candidates
+    uint256 public numCandidates;
+
+    /// @notice endorsements by a given voter
+    mapping(address => uint256[]) public endorsementsBy;
+
+    /// @notice the NFT contract for templars
+    Templar public templars;
+
+    /// @notice Nonces used in relayed voting requests
+    mapping(address => Counters.Counter) public _nonces;
```

Comment:
why name with underscore (_nonces) if it's public? thought that was primarily for private vars

Reply:
@butlerji Does _nonces need to be public?

Reply:
Happy for nonce to be public.

My 'default' is to make things public as it makes debugging easier, but use _ as a convention so other contributors know the _intent_ is to be private.

However, given the caller needs the next nonce, happy to drop the underscore and make this public :)